zetaclient/orchestrator/chain_activate.go

```diff
@@ -0,0 +1,248 @@
+package orchestrator
+
+import (
+	"time"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+
+	btcobserver "github.com/zeta-chain/zetacore/zetaclient/chains/bitcoin/observer"
+	btcrpc "github.com/zeta-chain/zetacore/zetaclient/chains/bitcoin/rpc"
+	btcsigner "github.com/zeta-chain/zetacore/zetaclient/chains/bitcoin/signer"
+	evmobserver "github.com/zeta-chain/zetacore/zetaclient/chains/evm/observer"
+	evmsigner "github.com/zeta-chain/zetacore/zetaclient/chains/evm/signer"
+	"github.com/zeta-chain/zetacore/zetaclient/chains/interfaces"
+	"github.com/zeta-chain/zetacore/zetaclient/common"
+	"github.com/zeta-chain/zetacore/zetaclient/config"
+)
+
+// WatchEnabledChains watches for run-time chain activation and deactivation
+func (oc *Orchestrator) WatchEnabledChains() {
+	oc.logger.Std.Info().Msg("WatchChainActivation started")
+
+	ticker := time.NewTicker(common.ZetaBlockTime * 2)
```

Comment:
Not sure how extensive is this action but I think we can have a larger ticker. Can be done every minutes for example

Reply:
It's a good consideration though. The number of `2` blocks I used was to not hurt (delay) local e2e tests by waiting too much time to activate the local BTC and Goerli chains. It can be special handled for local e2e test, the `2` blocks (12 seconds) seems probably a number work for local, testnet and mainnet.

How extensive:

- It triggers 2 `ethclient.Dial` for EVM chain (can be optimized in future) and 1 `rpcClient.Ping()` for BTC chain. Generally, 1 RPC dail per chain.
- It construct 2 maps of signers and observers in memory. In-memory operation is always fast than IO.

Reply:
Ok sounds good